src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Font.kt

```diff
@@ -0,0 +1,397 @@
+    private lateinit var boldData: Array<CharacterData?>
+    private lateinit var italicsData: Array<CharacterData?>
+    private var generatedCharData = false
+
+    private val colorCodes = IntArray(32)
+
+    init {
+        val inputStream = FileInputStream("${Config.modulesFolder}/$path")
+        this.font = Font.createFont(0, inputStream).deriveFont(Font.PLAIN, size)
+
+        Client.scheduleTask {
+            regularData = setup(arrayOfNulls(256), Font.PLAIN)
+            boldData = setup(arrayOfNulls(256), Font.BOLD)
+            italicsData = setup(arrayOfNulls(256), Font.ITALIC)
+
+            generatedCharData = true
+        }
+    }
+
+    private fun setup(characterData: Array<CharacterData?>, type: Int): Array<CharacterData?> {
+        generateColors()
+
+        val font = font.deriveFont(type)
+
+        val utilityImage = BufferedImage(1, 1, BufferedImage.TYPE_INT_ARGB)
+        val utilityGraphics = (utilityImage.graphics as Graphics2D).also { it.font = font }
+
+        val fontMetrics = utilityGraphics.fontMetrics
+
+        for (index in characterData.indices) {
+            val character = index.toChar()
+            val characterBounds = fontMetrics.getStringBounds(character.toString() + "", utilityGraphics)
+
+            val width = characterBounds.width.toFloat() + 2 * MARGIN
+            val height = characterBounds.height.toFloat()
+
+            val characterImage = BufferedImage(
+                MathHelper.ceiling_double_int(width.toDouble()),
+                MathHelper.ceiling_double_int(height.toDouble()),
+                BufferedImage.TYPE_INT_ARGB
+            )
+
+            val graphics = (characterImage.graphics as Graphics2D).also {
+                it.font = font
+                it.color = Color(255, 255, 255, 0)
+                it.fillRect(0, 0, characterImage.width, characterImage.height)
+                it.color = Color.WHITE
+            }
+
+            graphics.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON)
+            graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON)
+            graphics.setRenderingHint(RenderingHints.KEY_RENDERING, RenderingHints.VALUE_RENDER_QUALITY)
+            graphics.setRenderingHint(RenderingHints.KEY_FRACTIONALMETRICS, RenderingHints.VALUE_FRACTIONALMETRICS_ON)
+
+            graphics.drawString(character.toString() + "", MARGIN, fontMetrics.ascent)
+
+            val textureId = GlStateManager.generateTexture()
+            createTexture(textureId, characterImage)
+
+            characterData[index] = CharacterData(
+                characterImage.width.toFloat(),
+                characterImage.height.toFloat(),
+                textureId
+            )
+        }
+
+        return characterData
+    }
+
+    private fun createTexture(textureId: Int, image: BufferedImage) {
+        val pixels = IntArray(image.width * image.height)
```

Comment:
Maybe I'm misunderstanding, but I don't think this intermediate array is necessary, since `BufferedImage` supports individual pixel access